packages/commonwealth/client/scripts/state/api/threads/fetchReactionCounts.ts

```diff
@@ -0,0 +1,32 @@
+// import { useQuery } from '@tanstack/react-query';
+import axios from 'axios';
+import app from 'state';
+// import { ApiEndpoints } from 'state/api/config';
+
+// const THREAD_REACTION_COUNTS_STALE_TIME = 30 * 1_000; // 30 s
+
+interface FetchThreadReactionCountsProps {
+  // chainId: string; // this should be used when we fully migrate thread controller to react query
+  threadIds: number[];
+}
+
+const fetchThreadReactionCounts = async ({ threadIds }: FetchThreadReactionCountsProps) => {
+  return await axios.post(`${app.serverUrl()}/reactionsCounts`, {
+    thread_ids: threadIds,
+    active_address: app.user.activeAccount?.address,
+  }).then((response) => ([...response.data.result]));
+};
+
+// TODO: when we migrate the threads controller to react query, then we should also implement cache logic below
+// for this query. The reason why it was not implemented is because "reactive" code from react query wont work in
+// the non-reactive scope of threads controller.
```

Comment:
we can use react-query in non react components. Take a look on `gatedTopic.ts` where `queryClient` is used. Would that be helpful to move as much as we can to the react-query right now?

Reply:
yes, we can definitely do that but it won't be "reactive" (won't cause a re-render). Also, I won't recommend using react query in non react files often, thinking behind is, it destroys the whole "reactive" purpose and suit of react query (even tho it is supported)

> Would that be helpful to move as much as we can to the react-query right now?

I do agree it's better to migrate as much code as we can in general cases, but in this specific case, feels like a bit over-engineering (maybe I am wrong), eventually we will be getting rid of that state so I would suggest it's fine to leave it there.

Curious as to what you think, and happy to migrate this too if you think otherwise.

Reply:
yup I agree that using react-query in non-react scope should be the last resort. That's why I ask if this would help in this particular case. I am fine with your solution and yes we definitely should strive into moving logic into react-query asap to make it all reactive. Not necessarily in one shot.